source/components/Autocomplete.js

```diff
@@ -187,29 +191,46 @@ class AutocompleteBase extends Component<AutocompleteProps, State> {
       maxSelections != null ? selectedOptions.length < maxSelections : true;
     const areFilteredOptionsAvailable =
       filteredOptions && filteredOptions.length > 0;
-
+    let skipValueSelection = false;
     if (
       !maxSelections ||
       (canMoreOptionsBeSelected && areFilteredOptionsAvailable)
     ) {
-      if (!selectedOption) return;
-      const option = selectedOption.trim();
-      const optionCanBeSelected =
-        (selectedOptions.indexOf(option) < 0 && !multipleSameSelections) ||
-        multipleSameSelections;
-
-      if (option && optionCanBeSelected && isOpen) {
-        const newSelectedOptions = _.concat(selectedOptions, option);
-        this.selectionChanged(newSelectedOptions, event);
-        this.setState({ selectedOptions: newSelectedOptions, isOpen: false });
+      if (!selectedOption || !selectedOption.length) return;
+      const option = _.isString(selectedOption) ? selectedOption.trim() : selectedOption;
+      let newSelectedOptions: Array<any> = [];
```

Comment:
Why not `Array<string>`?

Reply:
Other selectedOptions in this component were already defined as type 'any' and hence i reused 'any'. As we are dealing with only strings here, you're right. I have pushed the fix for this.